app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,15 @@
+class SessionsController < ApplicationController
+  def create
+    auth_hash = request.env["omniauth.auth"]
+
+    user = User.create_from_google(env["omniauth.auth"])
+    session[:user_id] = user.id
+    redirect_to root_path
+  end
```

Comment:
There should probably also be some checks for invalid data or if the user goes directly to the callback URL.